stl/inc/filesystem

```diff
@@ -2151,35 +2153,39 @@ namespace filesystem {
         // [fs.dir.entry.mods], modifiers
         void assign(const filesystem::path& _Path_arg) {
             _Path.assign(_Path_arg);
-            (void) _Refresh(_Cached_data, _Path);
+            refresh();
         }
 
         void assign(const filesystem::path& _Path_arg, error_code& _Ec) {
             _Ec.clear(); // for exception safety
             _Path.assign(_Path_arg);
-            (void) _Refresh(_Cached_data, _Path);
+            refresh(_Ec);
         }
 
         void replace_filename(const filesystem::path& _Path_arg) {
             _Path.replace_filename(_Path_arg);
-            (void) _Refresh(_Cached_data, _Path);
+            refresh();

```

Comment:
This causes replace_filename(nonexistent) to throw which seems incorrect.

Reply:
I'm the original bug submitter and not the PR author. In my review of this PR, it looks like `refresh()` has been changed to never throw on "file not found".